### PR TITLE
tests/gnrc_netif_delayed_start: add test for delayed start of interfaces

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This tests application defined startup & shutdown of network interfaces.

Battery powered devices might only want to enable the radio if they have data to send.
For the most power saving and least startup time, that means they will not even configure the radio when not needed and just leave it in Deep Sleep.

For this to work the radio must remain in deep sleep across reboots and also allow to be shut down properly.

This adds a test to exercise that use-case.

### Testing procedure

Attach an ammeter to the supply line of your networking device.
 - run the `start` command to bring up your network interfaces. Current draw should be at `IDLE`/`RX` level.
  - run the `shutdown` command to put your network interfaces to sleep. Current draw should be at the lowest `SLEEP` level.
 - run the `reboot` command to confirm your interfaces remain sleeping across reboots

### CODE REVIEW

File: tests/gnrc_netif_delayed_start/Makefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+include ../Makefile.tests_common
+
+# Modules to include:
+USEMODULE += shell
+USEMODULE += shell_commands
+USEMODULE += ps
+
+USEMODULE += at86rf215
```
Comment (fjmolinas) [inline tests/gnrc_netif_delayed_start/Makefile:8]: Why is this needed? Shouldn't it be `USEMODULE += netdev_default`

Comment (benpicco) [reply tests/gnrc_netif_delayed_start/Makefile:8]: Ops sorry, I added that when I tested on `same54-xpro` and forgot to remove it again.

**Comment by fjmolinas** [comment]
Isn't there a way to query the radio state instead of relying on the ammeter measurements?

**Comment by benpicco** [comment]
You mean using `NETOPT_STATE`?
I can add that to see if the state was set correctly, but I wouldn't trust the radio driver for such a test - after all, this test is mainly to check if the radio driver is doing the right thing and the radio stays asleep.

**Comment by fjmolinas** [comment]
> You mean using NETOPT_STATE?
> I can add that to see if the state was set correctly, but I wouldn't trust the radio driver for such a test - after all, this test is mainly to check if the radio driver is doing the right thing and the radio stays asleep.

Its just that I'm not sure how useful this test is if its only manual, It would be better to somehow automatize this, otherwise it would never be run, and having a test checking the delayed start up functionality would be nice..

File: tests/gnrc_netif_delayed_start/Makefile (hunk 1 above)
Comment (fjmolinas) [reply tests/gnrc_netif_delayed_start/Makefile]: Still needs `USEMODULE += netdev_default`

**Comment by fjmolinas** [comment]
> Its just that I'm not sure how useful this test is if its only manual, It would be better to somehow automatize this, otherwise it would never be run, and having a test checking the delayed start up functionality would be nice..

Could you add an automatic test that would:

- send `ifconfig` expect it not to work
- send `start`, send `ifconfig` expect it to work
- check that `NETOPT` are set
- send `shutdown`, check that `NETOPT` are set

We can of course check current consumption or driver state at reboot, so just checking those should be OK and enough IMO.

**Comment by fjmolinas** [comment]
@benpicco are you not interested in this one anymore?
